client/core/core.go

```diff
@@ -1530,7 +1542,13 @@ func (c *Core) ReconfigureWallet(appPW []byte, assetID uint32, cfg map[string]st
 		return newError(addrErr, "error getting wallet address: %v", err)
 	}
 	dbWallet.Address = addr
-	wallet.address = addr
+	wallet.setAddress(addr)
+
+	// Must connect to ensure settings are good.
+	err = c.connectWallet(wallet)
```

Comment:
Without having tested these changes, I'm confused about the `ReconfigureWallet` edit because it would seem to be requesting a new deposit address from the wallet backend *before* connecting to the new wallet.

But I'm thinking that with the changes to `connectWallet` that ensures the current deposit address are owned by the wallet, that `ReconfigureWallet` no longer even needs to unconditionally get a new deposit address.

Reply:
Yep, that's true, but calling the `NewDepositAddress` inside the connectWallet needed the lock which caused 
a deadlock when generating new address, but as you described above, I would have already the xcWallet and can call the unexported method.